clash-prelude/src/Clash/Signal/Internal.hs

```diff
@@ -1436,6 +1436,30 @@ infix 4 .==.
 (.==.) :: (Eq a, Applicative f) => f a -> f a -> f Bool
 (.==.) = liftA2 (==)
 
+infix 4 .==
+-- | The above type is a generalization for:
+--
+-- @
+-- __(.==)__ :: 'Eq' a => 'Clash.Signal.Signal' a -> a -> 'Clash.Signal.Signal' 'Bool'
+-- @
+--
+-- It is a version of ('==') that allows comparing a @'Clash.Signal.Signal' a@ with a
+-- constant @a@ and returns a 'Clash.Signal.Signal' of 'Bool'
+(.==) :: (Eq a, Applicative f) => f a -> a -> f Bool
+(.==) a b = fmap (==b) a
```

Comment:
All of these new functions where only one argument is in a container don't need to be `Applicative`, only `Functor`. You hint at this in the implementations already: only `fmap` is needed to define these functions